packages/validator/src/util/params.ts

```diff
@@ -1,25 +1,61 @@
-import {IChainConfig, chainConfigToJson, defaultChainConfig} from "@chainsafe/lodestar-config";
+import {IChainConfig, chainConfigToJson} from "@chainsafe/lodestar-config";
+import {activePreset, BeaconPreset, presetToJson} from "@chainsafe/lodestar-params";
 
 export class NotEqualParamsError extends Error {}
 
+type ConfigWithPreset = IChainConfig & BeaconPreset;
+
+/* eslint-disable @typescript-eslint/naming-convention */
+
 /**
  * Assert localConfig values match externalSpecJson. externalSpecJson may contain more values than localConfig.
+ *
+ * This check ensures that the validator is connected to a beacon node of the exact same network and params.
+ * Otherwise, signatures may be rejected, time may be un-equal and other bugs that are harder to debug caused
+ * by different parameters.
+ *
+ * This check however can't compare the full config as is, since some parameters are not critical to the spec and
+ * can be changed un-expectedly. Also, fork parameters can change un-expectedly, like their _FORK_VERSION or _EPOCH.
+ * Note that the config API endpoint is not precisely specified, so each clients can return a different set of
+ * parameters.
+ *
+ * So this check only compares a specific list of parameters that are consensus critical, ignoring the rest. Typed
+ * config and preset ensure new parameters are leveled critical or ignore, faciliating manteinance of the list.
  */
 export function assertEqualParams(localConfig: IChainConfig, externalSpecJson: Record<string, string>): void {
```

Comment:
it would also be nice to update the tests for this logic with this new requirement